### PR TITLE
Update `Button` plain and plain destructive styles

### PR DESCRIPTION
**Pull request by aaronccasanova**
Closes https://github.com/Shopify/polaris-summer-editions/issues/105

[Storybook](https://5d559397bae39100201eedc1-uasspziesb.chromatic.com/?path=/story/all-components-button--all&globals=polarisSummerEditions2023:true)

Note: This PR adds new `color` tokens
- `color-text-critical-hover-experimental`
- `color-icon-critical-strong-hover-experimental`
- `color-icon-critical-strong-active-experimental`

### CODE REVIEW

**Review by lgriffee — APPROVED** [review]
LGTM 💯 (aside from the linting failures for combinators and nesting depth).

**Review by sarahill — APPROVED** [review]

**Review by kyledurand — APPROVED** [review]

**Review by sophschneider — APPROVED** [review]

File: polaris-react/src/components/Button/Button.scss [hunk 1]
```diff
@@ -648,6 +652,13 @@
     font-weight: var(--p-font-weight-regular);
     // stylelint-disable-next-line -- generated by polaris-migrator DO NOT COPY
     @include focus-ring;
+
+    #{$se23} &::after {
+      top: -2px;
```
Comment (aveline) [inline polaris-react/src/components/Button/Button.scss]: Might want to pull in the latest from `main` and double check if this is impacted by https://github.com/Shopify/polaris/pull/9401

File: polaris-react/src/components/Button/Button.scss [hunk 2]
```diff
@@ -767,94 +823,89 @@
...
+      }
     }
 
     &:hover svg {
       fill: var(--p-color-icon-hover);
     }
 
+    #{$se23} &:hover,
+    #{$se23} &:focus:not(.disabled) {
+      fill: var(--p-color-icon-interactive-hover);
```
Comment (sophschneider) [inline polaris-react/src/components/Button/Button.scss:849]: I think these `iconOnly` + `plain` background colors should be transparent

Screenshot 2023-06-08 at 12 10 24 PM

**Review by sophschneider — CHANGES_REQUESTED** [review]

**Comment by sarahill** [comment]
@aaronccasanova, @aveline made a [good point](https://github.com/Shopify/polaris-summer-editions/issues/105#issuecomment-1582860849) about the focus ring on plain buttons. I wanted to give these a little more affordance but they're used so interchangeably with links let's align the focus ring to links for consistency.

I updated Figma to match
image

File: polaris-react/src/components/Button/Button.scss (hunk 2 above)
Comment (sarahill) [reply polaris-react/src/components/Button/Button.scss:849]: It's hard for me to decipher the levels of nesting in the css to figure out which variant this is targeting but there are `plain` and `plain primary` variants. 

`plain` doesn't get the bg and `plain primary` does

Comment (sophschneider) [reply polaris-react/src/components/Button/Button.scss:849]: yeah its hard for me to decipher as well 😅 I tested on `plain` + `primary` locally and it didn't seem to have the transparent background

**Comment by aaronccasanova** [comment]
> let's align the focus ring to links for consistency

Done 👍 https://github.com/Shopify/polaris-summer-editions/issues/105#issuecomment-1583050412

File: polaris-react/src/components/Button/Button.scss (hunk 2 above)
Comment (aaronccasanova) [reply polaris-react/src/components/Button/Button.scss:849]: The style updates in this PR are scoped to `plain` and `plain destructive`

Comment (sophschneider) [reply polaris-react/src/components/Button/Button.scss:849]: okie, although [icon only](https://github.com/Shopify/polaris-summer-editions/issues/91) already got merged so maybe we should make a separate issue to track `plain` `iconOnly` ? cc: @aveline

**Review by sophschneider — APPROVED** [review]

File: polaris-react/src/components/Button/Button.scss (hunk 2 above)
Comment (sophschneider) [reply polaris-react/src/components/Button/Button.scss:849]: made a [new issue here](https://github.com/Shopify/polaris-summer-editions/issues/240)!

**Comment by sarahill** [comment]
> > let's align the focus ring to links for consistency
> 
> Done 👍 [Shopify/polaris-summer-editions#105 (comment)](https://github.com/Shopify/polaris-summer-editions/issues/105#issuecomment-1583050412)

@aaronccasanova  The padding still looks a bit bigger than on link focus. Can you confirm?

**Review by sarahill — APPROVED** [review]

**Pull request merged**